### ISSUE TITLE
Non-null doesn't seem to be recorded as the default for array elements

### ISSUE THREAD

**Issue by leavens** [OpenJML/OpenJML#719]
When I specify a method with an array formal, the elements of the array do not seem to be properly recorded as non-null by default for a generic type. For the attached program I get:
Compilation started at Wed Jan 13 13:13:40

esc Swap.java
Swap.java:10: warning: The prover cannot establish an assertion (PossiblyNullAssignment) in method <T>swap
          temp = arr[i];
               ^
1 warning

Compilation finished at Wed Jan 13 13:13:46
[Swap.zip](https://github.com/OpenJML/OpenJML/files/5810394/Swap.zip)

**Comment by leavens** [comment]
Just to note, this issue is not connected to generics, it also happens with Object as the type of the elements of the formal array parameter
[SwapObject.zip](https://github.com/OpenJML/OpenJML/files/5810410/SwapObject.zip)
, as in the attached.

**Comment by davidcok** [comment]
Correct. We should change to Ernst’s design for non-null annotations for arrays as adopted by Java for Java8 in order to support this cleanly.

**Comment by davidcok** [comment]
Confirmed fix of this issue. and test SFBugs.gitbug719
